### ISSUE TITLE
Support ISO 8601 date format in dates_times.dates.dash_and_from

### ISSUE THREAD

**Issue by cbachhuber** [amperser/proselint#1482]
### Expected Behaviour

When I write `from 2026-02-24 to 2026-02-25`, proselint should not complain.

### Actual Behaviour

It raises

```
<file>: dates_times.dates.dash_and_from: When specifying a date range, write 'from X to Y'.
```

### Reproduction

Put `from 2026-02-24 to 2026-02-25` into a file and let proselint lint the file. As soon as I remove the dashes, proselint doesn't complain anymore.

### Environment

Ubuntu 24.04 with proselint V0.16.0 via pre-commit

Possibly related to https://github.com/amperser/proselint/issues/341

**Comment by Nytelife26** [comment]
I believe this may be fixed trivially with boundary padding and a negative lookahead, but I will need to do further testing. Thank you for flagging this.
